calico-vpp-agent/cni/cni_server.go

```diff
@@ -378,9 +472,31 @@ func (s *Server) ServeCNI(t *tomb.Tomb) error {
 	}
 
 	pb.RegisterCniDataplaneServer(s.grpcServer, s)
+
+	netsSynced := make(chan bool)
+	nets := &[]networkv3.Network{}
+	go func() {
+		for t.Alive() {
+			event := <-s.cniEventChan
+			switch event.Type {
+			case common.NetsSynced:
+				nets = event.New.(*[]networkv3.Network)
```

Comment:
I think we can store the networks received in `nets` in `s.networkDefinitions`, which should remove the need for the `waitForNetsSynced` function.